### PR TITLE
Add periodic, with_positions to hexagonal_lattice_graph

### LINKED ISSUE #803: Add support to `hexagonal_lattice_graph` for periodcity and position coordinates
### What is the expected enhancement?
networkx's [`hexagonal_lattice_graph`](https://networkx.org/documentation/stable/reference/generated/networkx.generators.lattice.hexagonal_lattice_graph.html) function provides support for periodicity and setting position coordinates, and it would be great if rustworkx's `hexagonal_lattice_graph` did the same.

Definition of done:
- a `periodic` bool argument added to `hexagonal_lattice_graph`
- a `with_positions` bool argument added to `hexagonal_lattice_graph`

cc @mrossinek

### PR DESCRIPTION
**Pull request by WingCode**
closes #803

### CODE REVIEW

File: rustworkx-core/src/generators/hexagonal_lattice_graph.rs [hunk 1]
```diff
@@ -78,16 +78,24 @@ pub fn hexagonal_lattice_graph<G, T, F, H, M>(
     mut default_node_weight: F,
     mut default_edge_weight: H,
     bidirectional: bool,
+    periodic: bool,
+    with_positions: Option<F>,
```
Comment (IvanIsCoding) [inline rustworkx-core/src/generators/hexagonal_lattice_graph.rs:82]: This is a breaking API change on the Rust side. On the Python side, you can use optional arguments but on Rust there is not much we can do.

**Review by IvanIsCoding — COMMENTED** [review]
Thanks for the PR! I know your code does not compile yet, but I think we can get started on dicussing how to tackle the breaking change on the Rust side

**Comment by jpacold** [comment]
Hi @WingCode -- I noticed that this PR seems to only add `with_positions`, so I opened a separate draft PR just on the `periodic` part. Please let me know if you are planning to continue looking at this issue (I don't want to get in the way or duplicate your work).

**Comment by WingCode** [comment]
Hi @jpacold , Thank you for asking. Yes, you can go right ahead with issue. I am not sure when I will be able to get back to this issue.

**Comment by IvanIsCoding** [comment]
Closing this as obsolete after #1213. Nevertheless, thanks for getting the work started!

**Pull request closed without merging**